lib/eth/abi.rb

```diff
@@ -291,6 +291,17 @@ def decode_primitive_type(type, data)
       end
     end
 
+    # Build event signature string from ABI interface.
+    #
+    # @param interface [Hash] ABI event interface.
+    # @return [String] interface signature string.
+    def signature(interface)
+      name = interface.fetch("name")
```

Comment:
Do we want this to raise an exception if no name is found?

Reply:
I'd say so. An ABI without a name isn't valid. We can add a more specific exception class if you think that helps, `class InvalidInterfaceError < StandardError`.